packages/analytics-browser/src/attribution/campaign-tracker.ts

```diff
@@ -16,30 +15,39 @@ export class CampaignTracker implements ICampaignTracker {
   storage: Storage<Campaign>;
   storageKey: string;
   parser: ICampaignParser;
-  track: CampaignTrackFunction;
-  onNewCampaign: (campaign: Campaign) => unknown;
 
   disabled: boolean;
-  trackNewCampaigns: boolean;
-  trackPageViews: boolean;
   excludeReferrers: string[];
   initialEmptyValue: string;
+  resetSessionOnNewCampaign: boolean;
+
+  private _currentCampaign!: Campaign;
+  private _isNewCampaign!: boolean;
 
   constructor(apiKey: string, options: CampaignTrackerOptions) {
     this.storage = options.storage;
     this.storageKey = getStorageKey(apiKey, MKTG);
     this.parser = new CampaignParser();
-    this.track = options.track;
-    this.onNewCampaign = options.onNewCampaign;
 
     this.disabled = Boolean(options.disabled);
-    this.trackNewCampaigns = Boolean(options.trackNewCampaigns);
-    this.trackPageViews = Boolean(options.trackPageViews);
     this.excludeReferrers = options.excludeReferrers ?? [];
+    this.initialEmptyValue = options.initialEmptyValue ?? EMPTY_VALUE;
+    this.resetSessionOnNewCampaign = options.resetSessionOnNewCampaign ?? options.trackNewCampaigns === true;
+
     if (typeof location !== 'undefined') {
       this.excludeReferrers.unshift(location.hostname);
     }
-    this.initialEmptyValue = options.initialEmptyValue ?? EMPTY_VALUE;
+
+    void this.refreshCampaignState();
```

Comment:
Race condition - Track completion of refreshCampaignState in a promise field so getCurrentState can await it below.